controllers/configmap_controller.go

```diff
@@ -170,6 +177,58 @@ func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServ
 	return nil
 }
 
+// removeOutdatedServicesConfigMaps deletes any outdated services ConfigMaps, if all nodes have moved past that version
+func (r *ConfigMapReconciler) removeOutdatedServicesConfigMaps(ctx context.Context) error {
+	nodes := &core.NodeList{}
+	if err := r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return err
+	}
+	servicesConfigMaps, err := r.listServiceConfigMaps(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve list of services ConfigMaps")
+	}
+	for _, cm := range servicesConfigMaps {
+		if strings.HasSuffix(cm.Name, version.Get()) {
+			// Do not delete the services ConfigMap tied to the current WMCO version
+			continue
+		}
+		// Remove any services ConfigMap tied to a WMCO version that no Windows nodes are at anymore
+		if !isTiedToRelevantVersion(cm.Name, nodes.Items) {
+			if err := r.client.Delete(ctx, &cm); err != nil {
+				return errors.Wrapf(err, "could not delete outdated services ConfigMap %s", cm.Name)
+			}
+			r.log.Info("Deleted outdated resource", "ConfigMap",
+				kubeTypes.NamespacedName{Namespace: r.watchNamespace, Name: cm.Name})
```

Comment:
Instead of doing this, which can result in going through all the node objects multiple times i suggest
going through the nodes once, and getting a map[string]struct{} indicating all the current versions
you can then use `strings.TrimPrefix()` to get the CM version, and check the map for a match.